### PR TITLE
Add seasons to Rates table and CancellationPolicies table

### PR DESCRIPTION
**Pull request by oleg1std**

### CODE REVIEW

File: HappyTravel.Hiroshima.DirectContracts/Services/Availability/PaymentDetailsService.cs [hunk 1]
```diff
@@ -12,11 +12,11 @@ namespace HappyTravel.Hiroshima.DirectContracts.Services.Availability
     public class PaymentDetailsService: IPaymentDetailsService
     {
         public PaymentDetails Create(DateTime checkInDate, DateTime checkOutDate,
-            List<RoomRate> roomRates, List<RoomPromotionalOffer> roomPromotionalOffers)
+            List<RateDetails> roomRates, List<RoomPromotionalOffer> roomPromotionalOffers)
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Availability/PaymentDetailsService.cs:15]: `List<RateDetails> rateDetails`

File: HappyTravel.Hiroshima.Data/Migrations/DirectContractsDbContextModelSnapshot.cs [hunk 2]
```diff
@@ -240,20 +240,25 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
             modelBuilder.Entity("HappyTravel.Hiroshima.Data.Models.Location.Location", b =>
                 {
+                    b.Property<int>("Id")
+                        .ValueGeneratedOnAdd()
```
Comment (kirillta) [inline HappyTravel.Hiroshima.Data/Migrations/DirectContractsDbContextModelSnapshot.cs:244]: If I remember right, you have no need to define that conditions for primary keys

File: HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs [hunk 3]
```diff
@@ -84,25 +85,32 @@ public Task<List<AccommodationDetails>> GetAccommodations(string accommodationNa
                 .ToListAsync();
         }
 
-        
-        public async Task<List<RoomRate>> GetRates(IEnumerable<int> roomIds, DateTime checkInDate,
+        public async Task<List<RateDetails>> GetRates(IEnumerable<int> roomIds, DateTime checkInDate,
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:88]: Could `IEnumerable<int> roomIds` has narrower type definition?

File: HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs [hunk 4]
```diff
@@ -84,25 +85,32 @@ public Task<List<AccommodationDetails>> GetAccommodations(string accommodationNa
...
             DateTime checkOutDate, string languageCode)
         {
             checkInDate = checkInDate.Date;
             return await _dbContext.RoomRates
-                .Where(roomRate => roomIds.Contains(roomRate.RoomId) &&
-                                   !(roomRate.EndDate < checkInDate || checkOutDate < roomRate.StartDate))
-                .Select(roomRate => new RoomRate
+                .Join(_dbContext.Seasons, roomRate => roomRate.SeasonId, season => season.Id,
+                    (roomRate, season) => new {roomRate, season})
+                .Where(roomRateAndSeason => roomIds.Contains(roomRateAndSeason.roomRate.RoomId) &&
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:95]: Awkward formatting

File: HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs [hunk 5]
```diff
@@ -84,25 +85,32 @@ public Task<List<AccommodationDetails>> GetAccommodations(string accommodationNa
...
+                            DirectContractsDbContext.GetLangFromJsonb(roomRateAndSeason.roomRate.Details,
+                                languageCode),
+                        Price = roomRateAndSeason.roomRate.Price,
+                        BoardBasis = roomRateAndSeason.roomRate.BoardBasis,
+                        CurrencyCode = roomRateAndSeason.roomRate.CurrencyCode,
+                        SeasonId = roomRateAndSeason.roomRate.SeasonId,
+                        MealPlan = roomRateAndSeason.roomRate.MealPlan,
+                        RoomId = roomRateAndSeason.roomRate.RoomId
+                    },
+                    Season = roomRateAndSeason.season
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:113]: Why do we have `GetRates.SeasonId` and `Season` in one entity? This is a doubled responsibility

Comment (kirillta) [reply HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:113]: Also why do we extract a `Season` from the `GetRates` method?

File: HappyTravel.Hiroshima.Data/Migrations/DirectContractsDbContextModelSnapshot.cs (hunk 2 above)
Comment (oleg1std) [reply HappyTravel.Hiroshima.Data/Migrations/DirectContractsDbContextModelSnapshot.cs:244]: It's an auto-generated code

File: HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs (hunk 5 above)
Comment (oleg1std) [reply HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:113]: Rate and Season are DB models and they are related by season id

Comment (oleg1std) [reply HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:113]: Without a season we can not calculate validly an offer price

Comment (kirillta) [reply HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:113]: Then it must be separated to two queries or included in the `RoomRate`

Comment (kirillta) [reply HappyTravel.Hiroshima.DirectContracts/Services/Availability/AvailabilityRepository.cs:113]: You can't return two top-level objects from a repository query

**Pull request merged**